### PR TITLE
nmcli: Add macvlan connection type support

### LINKED ISSUE #5316: Support for interfaces of type MACVLAN using nmcli module.
### Summary

Currently there is no support for interfaces of type macvlan using the nmcli module.

Currently running a similar script will configure a macvlan that persists across reboots.

```script
nmcli connection add \
   type macvlan \
   con-name macvlan0
   dev eth0 \
   mode bridge \
   mac aa:bb:cc:00:11:22 \
   802-3-ethernet.cloned-mac-address aa:bb:cc:00:11:22 \
   ifname macvlan0 \
   ip4 192.168.1.1/24 \
   ipv6.method ignore \
   connection.autoconnect yes
```

### Issue Type

Feature Idea

### Component Name

nmcli

### Additional Information

```yaml (paste below)
name: Configure connection
  community.general.nmcli:
    type: "macvlan"
    conn_name: "macvlan0"
    ifname: "macvlan0"
    autoconnect: True
    ip4: "192.168.1.1/24"
    macvlanhwaddress: "aa:bb:cc:00:11:22"
    macvlandev: "eth0" 
    mode: "bridge"
    state: present
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by psvmcc**
##### SUMMARY

Add `macvlan` connection type support to `nmcli` module

Fixes #5316 

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

nmcli

### CODE REVIEW

**Review by russoz — COMMENTED** [review]
Hi @psvmcc Thanks for your contribution!

A couple of comments though.

File: plugins/modules/nmcli.py [hunk 1]
```diff
@@ -2342,6 +2398,7 @@ def main():
             wifi=dict(type='dict'),
             wifi_sec=dict(type='dict', no_log=True),
             gsm=dict(type='dict'),
+            macvlan=dict(type='dict'),
```
[the remark this answers is not in the document]
Comment (psvmcc) [reply plugins/modules/nmcli.py]: Added, please check now.

File: plugins/modules/nmcli.py [hunk 2]
```diff
@@ -879,6 +880,41 @@
...
+    macvlan:
+        description:
+            - The configuration of the MAC VLAN connection.
+            - Note the list of suboption attributes may vary depending on which version of NetworkManager/nmcli is installed on the host.
+            - 'An up-to-date list of supported attributes can be found here:
+              U(https://networkmanager.dev/docs/api/latest/settings-macvlan.html).'
+        type: dict
+        version_added: 6.6.0
+        suboptions:
+            mode:
```
[the remark this answers is not in the document]
Comment (psvmcc) [reply plugins/modules/nmcli.py:892]: Seems I'm was wrong here, after some rechecks I added `required` option to `mode` and `parent` params. Thanks!

File: plugins/modules/nmcli.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/nmcli.py]: looking good, thanks

File: plugins/modules/nmcli.py [hunk 3]
```diff
@@ -879,6 +880,41 @@
...
+                description:
+                    - The macvlan mode, which specifies the communication mechanism between multiple macvlans on the same lower device.
+                    - 'Following choices are allowed:
+                      C(0) B(unknown) (default)
+                      C(1) B(vepa)
+                      C(2) B(bridge)
+                      C(3) B(private)
+                      C(4) B(passthru)
+                      C(5) B(source)'
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/nmcli.py]: Yeah, probably not much options in this context.

**Comment by russoz** [comment]
maybe we could use `required_if=('state', 'macvlan', ['macvlan'])` in the module definition?

File: plugins/modules/nmcli.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (psvmcc) [reply plugins/modules/nmcli.py]: Done, please check.

**Comment by psvmcc** [comment]
> maybe we could use `required_if=('state', 'macvlan', ['macvlan'])` in the module definition?

@russoz if I correctly understand your idea, you suggest do next thing:
```
        mutually_exclusive=[['never_default4', 'gw4'],
                            ['routes4_extended', 'routes4'],
                            ['routes6_extended', 'routes6']],
-       required_if=[("type", "wifi", [("ssid")])],
+       required_if=[("type", "wifi", [("ssid")]),
+                    ("type", "macvlan", ["macvlan"])],
        supports_check_mode=True,
```
But in this case for absent state, will be required to set macvlan dict also, I think this is not good idea.

Or I'm wrong?

**Comment by russoz** [comment]
> > maybe we could use `required_if=('state', 'macvlan', ['macvlan'])` in the module definition?
> 
> @russoz if I correctly understand your idea, you suggest do next thing:
> ```
>         mutually_exclusive=[['never_default4', 'gw4'],
>                             ['routes4_extended', 'routes4'],
>                             ['routes6_extended', 'routes6']],
> -       required_if=[("type", "wifi", [("ssid")])],
> +       required_if=[("type", "wifi", [("ssid")]),
> +                    ("type", "macvlan", ["macvlan"])],
>         supports_check_mode=True,
> ```
> But in this case for absent state, will be required to set macvlan dict also, I think this is not good idea.
> 
> Or I'm wrong?

No, you're right on spot.

**Comment by psvmcc** [comment]
> > > maybe we could use `required_if=('state', 'macvlan', ['macvlan'])` in the module definition?
> > 
> > 
> > @russoz if I correctly understand your idea, you suggest do next thing:
> > ```
> >         mutually_exclusive=[['never_default4', 'gw4'],
> >                             ['routes4_extended', 'routes4'],
> >                             ['routes6_extended', 'routes6']],
> > -       required_if=[("type", "wifi", [("ssid")])],
> > +       required_if=[("type", "wifi", [("ssid")]),
> > +                    ("type", "macvlan", ["macvlan"])],
> >         supports_check_mode=True,
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But in this case for absent state, will be required to set macvlan dict also, I think this is not good idea.
> > Or I'm wrong?
> 
> No, you're right on spot. But then, it would be nice to have a neat message coming back to the user, in case they fail to provide the parameter - instead of ~an exception popping up~ being silently ignored.

Can you please check now, I just added raise error, if required params not set.

**Review by russoz — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@psvmcc thanks for your contribution!
@russoz thanks for reviewing!

**Comment by ganniterix** [comment]
Thanks for this contribution!

**Pull request merged**